Modules/Filtering/Thresholding/include/itkIntermodesThresholdImageFilter.h

```diff
@@ -90,61 +90,57 @@ class IntermodesThresholdImageFilter : public HistogramThresholdImageFilter<TInp
   using MaskIndexType = typename MaskImageType::IndexType;
   using MaskImageRegionType = typename MaskImageType::RegionType;
 
-  using HistogramType = typename Superclass::HistogramType;
+  using ValueType = typename NumericTraits<InputPixelType>::ValueType;
+  using ValueRealType = typename NumericTraits<ValueType>::RealType;
+  using HistogramType = Statistics::Histogram<ValueRealType>;
+  using HistogramPointer = typename HistogramType::Pointer;
+  using HistogramConstPointer = typename HistogramType::ConstPointer;
+  using HistogramSizeType = typename HistogramType::SizeType;
+  using HistogramMeasurementType = typename HistogramType::MeasurementType;
+  using HistogramMeasurementVectorType = typename HistogramType::MeasurementVectorType;
+
   using CalculatorType = IntermodesThresholdCalculator<HistogramType, InputPixelType>;
 
+  /** Set/Get the calculator to use to compute the threshold */
+  itkSetObjectMacro(Calculator, CalculatorType);
+  itkGetModifiableObjectMacro(Calculator, CalculatorType);
```

Comment:
The parent class has a public Set/Get Calculator method. This method should be overloaded. The set method likely needs a dynamic_cast of the calculator.

Reply:
Consider the alternative solution of overloading these methods so that HistogramThresholdImageFilter::m_Calclulcator and this::m_ItermodeCalculator point to the same instance.

The overloaded HistogramThresholdImageFilter::SetCalculator method would need to dynamic_cast to the Intermodes calculator, the call HistogramThresholdImageFilter::SetCalculator and this::SetCalculator with the dynamic casted object.

Reply:
Not sure how you would create the `m_Calculator` object in [here](https://github.com/InsightSoftwareConsortium/ITK/pull/1718/files#diff-a4eb676b8dccbc553855aabae3ec343fR34), though

Reply:
HistogramThresholdImageFilter still have its m_Calculator ivar.
IntermodesThresholdImageFilter maintains its original m_IntermodesCalculator with derived type.

Both SetCalculator methods would set both variables.

Reply:
Thanks for the comments Brad. Please check commit 624c503. Especially the change in [here](https://github.com/InsightSoftwareConsortium/ITK/pull/1718/files#diff-e8fefc8c747bbff3a35dbb5c9fbfe84eL116) as I did not have a clear criterion to continue using an `itkGetModifiableObjectMacro` macro, and the base class `GetCalculator` implementation in [here](https://github.com/InsightSoftwareConsortium/ITK/pull/1718/files#diff-acc13746dfc9bfba2ffc93d33947b390R186).